molgenis-data-security/src/main/java/org/molgenis/data/security/audit/EntityTypeRepositoryAuditDecorator.java

```diff
@@ -0,0 +1,88 @@
+package org.molgenis.data.security.audit;
+
+import static com.google.common.collect.Streams.stream;
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.data.security.audit.AuthenticationUtils.getUsername;
+import static org.molgenis.data.semantic.Relation.isAudited;
+import static org.molgenis.data.semantic.Vocabulary.AUDIT_USAGE;
+
+import java.util.HashMap;
+import java.util.stream.Stream;
+import org.molgenis.audit.AuditEvent;
+import org.molgenis.audit.AuditEventPublisher;
+import org.molgenis.data.AbstractRepositoryDecorator;
+import org.molgenis.data.Repository;
+import org.molgenis.data.meta.model.EntityType;
+import org.molgenis.data.meta.model.Tag;
+
+/** Publishes {@link AuditEvent}s when an {@link EntityType}'s auditing is enabled or disabled. */
+public class EntityTypeRepositoryAuditDecorator extends AbstractRepositoryDecorator<EntityType> {
+
+  static final String AUDIT_ENABLED = "AUDIT_ENABLED";
```

Comment:
Maybe add the word EntityType here, the system wide audit settings may disable audit after all.